### ISSUE TITLE
PackageInfo: Invalid constraint (PyYAML (>=5.1.*)) found in omegaconf-2.1.0 dependencies, skipping

### ISSUE THREAD

**Issue by gygabyte017** [python-poetry/poetry#4206]
- [X] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [X] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [X] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: Windows 10
- **Poetry version**:  1.2.0a

## Issue
In any project, the command:

    poetry add omegaconf

leads to this error:

    PackageInfo: Invalid constraint (PyYAML (>=5.1.*)) found in omegaconf-2.1.0 dependencies, skipping

so that the dependency `pyyaml` is not installed and therefore the package `omegaconf` will not work.

However with `pip install omegaconf` the package is installed correctly together with its dependency `pyyaml`. What may be wrong?

**Comment by finswimmer** [comment]
Hello @gygabyte017,

this was reported earlier in:

 * https://github.com/python-poetry/poetry/issues/4201
 * https://github.com/python-poetry/poetry/issues/4095

TL;DR: `>=5.1.*` is invalid and should be fixed by the maintainer of the package to `>=5.1`

fin swimmer

**Comment by HaukurPall** [comment]
Is there a workaround? I use Fairseq (https://github.com/facebookresearch/fairseq/) which depends on this broken omegaconf specification. This project is not released frequently and is heavily integrated into this version of omegaconf.

**Comment by MeWannaSleep** [comment]
@HaukurPall Hi, Have you find any workaround?

**Comment by HaukurPall** [comment]
@MeWannaSleep instead of using poetry I used pip :)

**Comment by SakarDev** [comment]
> @MeWannaSleep instead of using poetry I used pip :)

But i already use pip and still getting same error!
`pip install fairseq`

**Comment by dimbleby** [comment]
you are using a closed issue on the poetry bug tracker to report an issue that you see when not using poetry.  This is absurd.

I expect more recent pip is parsing version specifiers more accurately - and therefore now agrees with poetry that the invalid constraint is indeed invalid.  Take your bug report to the project that provides the bad constraint.
